python_modules/dagster/dagster_tests/general_tests/test_checked.py

```diff
@@ -0,0 +1,151 @@
+from typing import TYPE_CHECKING, Annotated, Optional
+
+import pytest
+from dagster._check import CheckError
+from dagster._check.builder import ImportFrom
+from dagster._check.decorator import checked
+
+if TYPE_CHECKING:
+    from dagster._core.test_utils import TestType
+
+
+def test_basic():
+    @checked
+    def foo(): ...
+
+    foo()
+
+    @checked
+    def bar(i: int): ...
+
+    bar(1)
+    bar(i=1)
+    with pytest.raises(CheckError):
+        bar("1")  # type: ignore
+    with pytest.raises(CheckError):
+        bar(i="1")  # type: ignore
+
+
+class Thing: ...
+
+
+def test_many():
+    @checked
+    def big(
+        name: str,
+        nick_names: list[str],
+        age: int,
+        cool: bool,
+        thing: Optional[Thing],
+        other_thing: Thing,
+        percent: float,
+        o_s: Optional[str],
+        o_n: Optional[int],
+        o_f: Optional[float],
+        o_b: Optional[bool],
+        foos: list[Annotated["TestType", ImportFrom("dagster._core.test_utils")]],
```

Comment:
Talking about this, this actually feels worse than doing

```suggestion
        foos: "dagster._core.test_utils.TestType",
```

This seems much more standard/PEP-compliant.